packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -362,6 +363,17 @@ export class ForkChoice implements IForkChoice {
     // Update finalized checkpoint.
     if (finalizedCheckpoint.epoch > this.fcStore.finalizedCheckpoint.epoch) {
       this.fcStore.finalizedCheckpoint = toCheckpointWithHex(finalizedCheckpoint);
+      if (
+        !this.isMergeTransitionFinalized &&
+        bellatrix.isBellatrixStateType(state) &&
+        bellatrix.isBellatrixBlockBodyType(block.body) &&
+        bellatrix.isExecutionEnabled(state, block.body)
+      ) {
+        const finalizedBlock = this.getFinalizedBlock();
+        if (finalizedBlock.executionStatus !== ExecutionStatus.PreMerge) {
+          this.isMergeTransitionFinalized = true;
```

Comment:
Is this strategy sufficient to not log again after restarting?

Reply:
Moving this logic here just for the log doesn't feel great. Is lighthouse doing the same?

Reply:
lighthouse doesn't seem to be logging finalized, i wanted to add this on my own. 
on restart, if if it starts post merge, isMergeTransitionFinalized should be set true on init and shouldn't log.

Reply:
over here: https://github.com/ChainSafe/lodestar/pull/3963/files#diff-a941ba900b315f252132684ebd4701cf9b4efa664c99112856e3d0313a4ceee5R79

Reply:
While I know it's not a lot, I'm not super keen on adding _some_ complexity for a non critical item like this. While we (coredevs) celebrate finalization users may only care about the merge block, and it's very easy to detect once. Let me know what you think.

Reply:
I also have no solicited any banner work for finalized at this time.

Reply:
ok no issues, removed :slightly_smiling_face: